index.tt

```diff
@@ -6,152 +6,349 @@
-        approach to system configuration management to networks and
-        provisioning.
-      </p>
-      <a class="learn-more" href="https://github.com/NixOS/nixops">
-        More about NixOps
-        <i class="fa fa-angle-right"></i>
-      </a>
+  </section>
+  <section class="tab-pane" id="build">
+    <div class="row-fluid">
+      <div class="span6">
+        <h3>Minimal docker image</h3>
+        <p>
+          Declarative way to build minimal docker images. No build tools inside
+          docker image, no complex multi stage build process, only what your
+          application needs.
+        </p>
+        <p>
+          The following Nix expression (<code>default.nix</code>) defines a
+          docker image with <strong>only</strong> Python 3 installed in it.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.dockerTools.buildLayeredImage {
+  name = "only-python";
+  contents = [ pkgs.python3 ];
+}</pre>
+        <p>To build and run the image you need to:</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-build</strong></span>
+...
+/nix/store/hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3-docker-image-only-python.tar.gz
+<span class="nix-command">docker load -u ./result</span>
+...
+Loaded image: hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3
+<span class="nix-command">docker run hello:hb8xwmgd1pfrwgmc3indi2rrgcjr3wa3 python -c "print('hello world')"</span>
+hello world</pre>
+        <p>
+          Learn more <a href="[%root%]nixpkgs/manual/#sec-pkgs-dockerTools">how
+            to build docker images</a>.
+        </p>
+
+      </div>
+      <div class="span6">
+
+        <h3>Declarative cloud images</h3>
+        <p>How hard would it be to build and configure <strong>Amazon AMI?</strong></p>
+        <p>
+          With the following <code>amazon.nix</code> we defined nginx which is
+          serving example <code>/var/www</code> folder, having a valid ssl
+          certificate (via LetsEncrypt) and enabled recommended security
+          settings.
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs, ...}:
+{
+  security.acme.acceptTerms = true;
+  security.acme.email = "nix@example.com";
+  services.nginx = {
+    enable = true;
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+    virtualHosts."example.com" = {
+      enableACME = true;
+      forceSSL = true;
+      locations."/".root = "/var/www";
+    };
+  };
+}</pre>
+        <p>Now we just need to build it and upload it.</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-build '&lt;nixpkgs/nixos/release.nix&gt;' -A amazonImage.x86_64-linux --arg configuration ./amazon.nix</strong></span>
+...
+/nix/store/wmg6gcgnh6bk8d98syydfq73q6hjv682-nixos-amazon-image-20.09pre130979.gfedcba-x86_64-linux
```

Comment:
promote stable channel instead? i think it should be used in production environments

Reply:
I will hide which channel do we use and also all the hashes. this should also make it fit to the code block and not overflow.